dvc/fs/dvc.py

```diff
@@ -296,6 +297,11 @@ def get_file(  # pylint: disable=arguments-differ
         key = self._get_key_from_relative(rpath)
         fs_path = self._from_key(key)
         fs = self.repo.fs
+
+        if self.isdir(rpath):
+            makedirs(lpath, exist_ok=True)
+            return None
+
         try:
             fs.get_file(fs_path, lpath, callback=callback, **kwargs)
```

Comment:
Since we use `dvc.fs.data.DataFileSystem` which is not fsspec-compatible and `Repo.fs` which is also not fsspec compatible, this may likely cause us some problems/incompatibilities in the future.